app/controllers/application_controller.rb

```diff
@@ -22,6 +22,39 @@ class ApplicationController < ActionController::API
 
   private
 
+  set_current_tenant_through_filter
+  before_action :set_the_current_tenant
+  after_action  :unset_the_current_tenant
```

Comment:
This might be better designed as an around_action since they are a joined pair.

Reply:
I tried to do it that way but it seemed to bypass Brandon's code that checks parameters so I backed it out. Let me try again and maybe we can look at it together

Reply:
I think I can get this to work. Looking at it now